src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml

```diff
@@ -1785,7 +1786,7 @@ let pickles_compile (choices : pickles_rule_js Js.js_array Js.t)
         Public_input.(public_input_js |> of_js |> to_constant)
       in
       prover ?handler:None prevs public_input
-      |> Promise.map ~f:(fun ((), proof) -> proof)
+      |> Promise.map ~f:(fun ((), (), proof) -> proof)
```

Comment:
you also have to change the return type of the `main` function in `create_pickles_rule`, further above
unfortunately my type cheating prevents the type-checker from catching that

Reply:
I guess I'll unwind some of the type hacking. At some point, we should really remove it all, since it's such a foot-gun.

Reply:
with this I think it'll be fine: https://github.com/MinaProtocol/mina/pull/11018/commits/6ac53363a9424b3b658f6ab9d2ea9e5c06b9aa9e
we should merge it :)